src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs

```diff
@@ -1209,7 +1221,17 @@ private static TiktokenTokenizer CreateForModel(
                     tiktokenConfiguration.VocabFile,
                     out (Dictionary<ReadOnlyMemory<byte>, int> encoder, Dictionary<StringSpanOrdinalKey, (int Id, string Token)> vocab, Dictionary<int, ReadOnlyMemory<byte>> decoder) cache))
             {
-                using Stream compressedStream = typeof(Tokenizer).Assembly.GetManifestResourceStream(tiktokenConfiguration.VocabFile)!;
+                Debug.Assert(tiktokenConfiguration.TypeName is not null && tiktokenConfiguration.PackageName is not null);
+
+                string fullAssemblyName = typeof(TiktokenTokenizer).Assembly.FullName!;
+                int commaIndex = fullAssemblyName.IndexOf(',');
+                Type? type = commaIndex > 0 ? Type.GetType($"{tiktokenConfiguration.TypeName}, {tiktokenConfiguration.PackageName}{fullAssemblyName.Substring(commaIndex)}") : null;
```

Comment:
Is there a reason we use GetType instead of Assembly.Load?  I think we could get rid of all the empty types if we just called Load on the full name.

Reply:
Just realized you said the same in chat